libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -643,17 +656,18 @@ void AP_Airspeed::read(uint8_t i)
         return;
     }
 
-    bool prev_healthy = state[i].healthy;
     float raw_pressure = get_pressure(i);
-    if (state[i].cal.start_ms != 0) {
-        update_calibration(i, raw_pressure);
-    }
-
-    float airspeed_pressure = raw_pressure - param[i].offset;
+    float airspeed_pressure = raw_pressure - get_offset(i);
 
     // remember raw pressure for logging
     state[i].corrected_pressure = airspeed_pressure;
 
+#ifndef HAL_BUILD_AP_PERIPH
+    bool prev_healthy = state[i].healthy;
+    if (state[i].cal.start_ms != 0) {
+        update_calibration(i, raw_pressure);
```

Comment:
This slight re-ordering does mean that the calibrated offset does not get used to the next loop after a successful calibration.